# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureRecorded.h
# Commit: 31935f164ef0
# Full Hash: 31935f164ef0d6c4668f211482c139847dcab7f2
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-13 00:43:26
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 13c9bf69085f -r 31935f164ef0 gfx/layers/client/TextureRecorded.h
--- a/gfx/layers/client/TextureRecorded.h	Tue Dec 12 07:35:02 2023 +0000
+++ b/gfx/layers/client/TextureRecorded.h	Tue Dec 12 07:35:03 2023 +0000
@@ -32,6 +32,8 @@
 
   already_AddRefed<gfx::SourceSurface> BorrowSnapshot() final;
 
+  void ReturnSnapshot(already_AddRefed<gfx::SourceSurface> aSnapshot) final;
+
   void Deallocate(LayersIPCChannel* aAllocator) final;
 
   bool Serialize(SurfaceDescriptor& aDescriptor) final;
@@ -40,6 +42,11 @@
 
   TextureFlags GetTextureFlags() const final;
 
+  void SetRemoteTextureOwnerId(
+      RemoteTextureOwnerId aRemoteTextureOwnerId) final;
+
+  bool RequiresRefresh() const final;
+
  private:
   DISALLOW_COPY_AND_ASSIGN(RecordedTextureData);
 
@@ -51,7 +58,10 @@
   gfx::SurfaceFormat mFormat;
   RefPtr<gfx::DrawTarget> mDT;
   RefPtr<gfx::SourceSurface> mSnapshot;
+  ThreadSafeWeakPtr<gfx::SourceSurface> mSnapshotWrapper;
   OpenMode mLockedMode;
+  layers::RemoteTextureId mLastRemoteTextureId;
+  layers::RemoteTextureOwnerId mRemoteTextureOwnerId;
 };
 
 }  // namespace layers